src/main/java/io/cdap/plugin/http/source/common/http/SSLConnectionSocketFactoryCreator.java

```diff
@@ -0,0 +1,130 @@
+import javax.net.ssl.TrustManagerFactory;
+
+/**
+ * Class which creates an SSLConnectionSocketFactory.
+ */
+public class SSLConnectionSocketFactoryCreator {
+  private static final Logger LOG = LoggerFactory.getLogger(SSLConnectionSocketFactoryCreator.class);
+
+  private final BaseHttpSourceConfig config;
+
+  public SSLConnectionSocketFactoryCreator(BaseHttpSourceConfig config) {
+    this.config = config;
+  }
+
+  public SSLConnectionSocketFactory create() {
+    String cipherSuitesString = config.getCipherSuites();
+    // Usually colons are used as separators, but commas or spaces are also acceptable.
+    String[] cipherSuites = (cipherSuitesString == null) ? null : cipherSuitesString.split("[:, ]");
+
+    try {
+      SSLContext sslContext = SSLContext.getInstance("TLS"); // "TLS" means rely system properties
+      sslContext.init(getKeyManagers(), getTrustManagers(), null);
+
+      return new SSLConnectionSocketFactory(sslContext, config.getTransportProtocolsList().toArray(new String[0]),
+                                            cipherSuites, SSLConnectionSocketFactory.getDefaultHostnameVerifier());
+    } catch (KeyManagementException | CertificateException | NoSuchAlgorithmException | KeyStoreException
+      | IOException | UnrecoverableKeyException e) {
+      throw new IllegalStateException("Failed to create an SSL connection factory.", e);
+    }
+  }
+
+  private KeyManager[] getKeyManagers() throws CertificateException, NoSuchAlgorithmException,
+    KeyStoreException, IOException, UnrecoverableKeyException {
+
+    KeyStore keystore = loadKeystore(config.getKeystoreFile(), config.getKeystoreType().name(),
+                                     config.getKeystorePassword());
+
+    String keyStorePassword = config.getKeystorePassword();
+
+    // we have to manually fall back to default keystore. SSLContext won't provide such a functionality.
+    if (keystore == null) {
+      String keyStore = System.getProperty("javax.net.ssl.keyStore");
+      String keyStoreType = System.getProperty("javax.net.ssl.keyStoreType", KeyStore.getDefaultType());
+      keyStorePassword = System.getProperty("javax.net.ssl.keyStorePassword", "");
+
+      keystore = loadKeystore(keyStore, keyStoreType, keyStorePassword);
+    }
+
+    String keystoreAlgorithm =
+      (Strings.isNullOrEmpty(config.getKeystoreKeyAlgorithm())) ? KeyManagerFactory.getDefaultAlgorithm()
+        : config.getKeystoreKeyAlgorithm();
+    KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(keystoreAlgorithm);
+    char[] passwordArr = (keyStorePassword == null) ? null : keyStorePassword.toCharArray();
+    keyManagerFactory.init(keystore, passwordArr);
+    return keyManagerFactory.getKeyManagers();
+  }
+
+  private TrustManager[] getTrustManagers()
+    throws CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+
+    if (config.getVerifyHttps()) {
+      return new TrustManager[] { new TrustAllTrustManager() };
```

Comment:
it looks strange when we trust all certificates if varifyHttps is true

Reply:
thanks. Nice find.

Reply:
done